src/PostgREST/Auth.hs

```diff
@@ -163,14 +164,25 @@ middleware appState app req respond = do
 -- | Used to retrieve and insert JWT to JWT Cache
 getJWTFromCache :: AppState -> ByteString -> Int -> IO (Either Error AuthResult) -> UTCTime -> IO (Either Error AuthResult)
 getJWTFromCache appState token maxLifetime parseJwt utc = do
-  checkCache <- C.lookup (getJwtCache appState) token
+
+  -- Purge expired tokens in a separate thread
+  _ <- forkIO $ C.purgeExpired jwtCache
```

Comment:
Hm, this is going to start a new thread on every request? That doesn't look efficient.

I think we should have a background thread that is notified for doing the purge.

Reply:
To be honest - for starters I would implement a simple solution that executes purgeExpired periodically ( frequency subject to configuration possibly - but not necessarily ).

Triggering purge upon every request might mean the purging thread is running constantly wasting cycles as there is nothing to do most of the times.

The ultimate solution would be to have some kind of scheduler that triggers purging upon next nearest expiry - but that would require more complex bookkeeping.

Reply:
Thinking about it some more - it seems to me the best moment to trigger purge is upon a **cache miss** - just before/after inserting a new entry.

The reasoning is that:

1. We expect it to be rare (otherwise there is no point of the cache)
2. It makes sure the cache is not growing (as inserting new entries does garbage collection)
3. Since this is time expiration based cache there is no real risk of starvation - sooner or later we are going to have a cache miss.

Reply:
This seems like a much better idea. This way, we wouldn't need to do maintain any new thread state, hence avoiding extra overhead. @steve-chavez WDYT?

Reply:
Agree, looks like great idea :+1:

Reply:
@taimoorzaeem not sure how to help with that - do you think it might be ready soon? Seems to me without it JWT caching is kind of useless (ie. has memory leak). I think it is a pretty good candidate for quick patch release.

Reply:
@mkleczek Yes, I agree that JWT caching is not production ready yet. @steve-chavez WDYT?

Reply:
Yes, we should definitely do a patch release. We need to merge https://github.com/PostgREST/postgrest/pull/3802 first for testing, I'll review it tomorrow.